sklearn/utils/sparsefuncs.py

```diff
@@ -25,6 +26,53 @@ def _raise_error_wrong_axis(axis):
         raise ValueError(
             "Unknown axis value: %d. Use 0 for rows, or 1 for columns" % axis)
 
+def csr_csc_min_axis0(X):
+    """Compute the minimum along axis 0 for a csr or csc matrix.
+
+    Parameters
+    ----------
+    X: {csr_matrix, csc_matrix}, shape (n_samples, n_features)
+        Input data.
+
+    Returns
+    -------
+    minimum : array, shape [n_features]
+        The minimum of elements along axis 0.
+    """
+
+    # TODO: I don't know how to avoid this call to `astype()`;
+    # my test cases don't pass with integers otherwise.
+    X = X.astype(float)
+    if isinstance(X, sp.csr_matrix):
+        return csr_min_axis0(X)
+    elif isinstance(X, sp.csc_matrix):
+        return csc_min_axis0(X)
+    else:
+        raise ValueError("Input array was neither a csr_matrix nor csc_matrix.")
+
+def csr_csc_max_axis0(X):
+    """Compute the maximum along axis 0 for a csr or csc matrix.
+
+    Parameters
+    ----------
+    X: {csr_matrix, csc_matrix}, shape (n_samples, n_features)
+        Input data.
+
+    Returns
+    -------
+    maximum : array, shape [n_features]
+        The maximum of elements along axis 0.
+    """
+
+    # TODO: I don't know how to avoid this call to `astype()`;
+    # my test cases don't pass with integers otherwise.
+    X = X.astype(float)
```

Comment:
This is type of conversion is necessary because the Cython code doesn't do the conversion for us. I would use `np.float64` instead of `float` to be more explicit. This is the type that corresponds to `double`.